app/scripts/models/account.js

```diff
@@ -406,6 +406,24 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Check if user exists by uid.
+     *
+     * @returns {promise} - resolves when complete
+     */
+    checkAccountExists: function (uid) {
```

Comment:
An account can contain a `uid` or `email` as state, instead of passing the functions in, I'm wondering we would be better off doing something like setting the uid/email on the account, then using that state to check whether the UID or email exists:

```js
var account = new Account({ uid: user_id });
return account.checkAccountUidExists()
  .then(function (exists) {
    // do something here.
  });
```

or
```js
var account = new Account({ email: email });
return account.checkAccountEmailExists()
  .then(function (exists) {
    // do something here.
  }
```